app/models/device.rb

```diff
@@ -40,7 +40,8 @@ def ios_notification(title, body, payload)
           subtitle: title,
           body: body
```

Comment:
**non-blocking suggestion:** Since we're now passing in (possibly) a longer text for the body it might make sense to limit the length of it here to avoid going over the PN payload limit. If we add the restriction here it will cover all PNs implemented in the future, instead of limiting each one individually.

Maybe `.truncate(300)` or something along those lines would do the trick.

Reply:
Great point! iOS docs say the entire payload is limited to 4kb (4096 bytes), but there's nothing about each field. I found [this article](https://help.pushwoosh.com/hc/en-us/articles/360000440366-The-limit-of-characters-that-can-be-sent-through-a-push-notification) that indicates there's a limit of 178 characters displayed on the notification short look (the OS does this truncation), and seemingly no limit on the upper end.

UTF-8 is 4 bytes per character, so setting a limit of 2kb for the body field (2048 bytes) gives a max body size of 512 characters. I'll truncate to that!